src/Psalm/Checker/Statements/Expression/Fetch/ArrayFetchChecker.php

```diff
@@ -409,11 +421,18 @@ public static function getArrayAccessTypeGivenOffset(
                                 $offset_type
                             );
 
+                            $property_count = $type->sealed ? count($type->properties) : null;
+
                             $type = new TArray([
                                 $new_key_type,
                                 $generic_params,
                             ]);
 
+                            if (!$stmt->dim && $property_count) {
```

Comment:
How are arrays of 0 length handled?

Why is count treated as a property instead of something calculated from the array fields (and whether or not they are possibly undefined)

Reply:
https://github.com/vimeo/psalm/blob/1a4ddc7cdd1b332e04405a11db2b5227e4b1587e/src/Psalm/Type.php#L689

The issue suppression is redundant now, will remove